packages/lib/src/spatial-navigation/components/virtualizedGrid/SpatialNavigationVirtualizedGrid.test.tsx

```diff
@@ -0,0 +1,152 @@
+import { RenderResult, act, render, screen } from '@testing-library/react-native';
+import { ItemWithIndex } from '../virtualizedList/VirtualizedList';
+import { PropsTestButton, TestButton } from '../tests/TestButton';
+import { SpatialNavigationRoot } from '../Root';
+import '../tests/helpers/configureTestRemoteControl';
+
+import { DefaultFocus } from '../../context/DefaultFocusContext';
+import { SpatialNavigationVirtualizedGrid } from '../virtualizedGrid/SpatialNavigationVirtualizedGrid';
+import testRemoteControlManager from '../tests/helpers/testRemoteControlManager';
+
+export const expectButtonToHaveFocus = (component: RenderResult, text: string) => {
+  const element = component.getByRole('button', { name: text });
+  expect(element).toBeSelected();
+};
+
+describe('SpatialNavigationVirtualizedGrid', () => {
+  const renderItem = ({ item }: { item: PropsTestButton & ItemWithIndex }) => (
+    <TestButton title={item.title} onSelect={item.onSelect} />
+  );
+
+  function createDataArray(numberOfItems: number) {
+    const data = [];
+    for (let i = 0; i < numberOfItems; i++) {
+      data.push({
+        title: `button ${i + 1}`,
+        onSelect: () => undefined,
+        index: i,
+      });
+    }
+    return data;
+  }
+
+  const renderGrid = () =>
+    render(
+      <SpatialNavigationRoot>
+        <DefaultFocus>
+          <SpatialNavigationVirtualizedGrid
+            renderItem={renderItem}
+            data={createDataArray(19)}
+            itemHeight={100}
+            numberOfRenderedRows={5}
+            numberOfRowsVisibleOnScreen={3}
+            numberOfColumns={3}
+            testID="test-list"
```

Comment:
It could have been named differently but it's okay 😁 

```suggestion
            testID="test-grid"
```